lib/stripe/stripe_client.rb

```diff
@@ -121,15 +127,22 @@ def self.sleep_time(num_retries)
     #     charge, resp = client.request { Charge.create }
     #
     def request
-      @last_response = nil
-      old_stripe_client = Thread.current[:stripe_client]
-      Thread.current[:stripe_client] = self
+      old_stripe_client = self.class.current_thread_context.active_client
+      self.class.current_thread_context.active_client = self
+
+      if self.class.current_thread_context.last_responses&.key?(object_id)
```

Comment:
What do you think about the use of the safe navigation operator? Added in 2.3, which is technically our minimum version now.

Reply:
👍